src/main/java/org/apache/maven/plugins/gpg/GpgVersion.java

```diff
@@ -32,54 +32,46 @@
  */
 public class GpgVersion implements Comparable<GpgVersion>
 {
-    private final String rawVersion;
 
-    private GpgVersion( String rawVersion )
-    {
-        this.rawVersion = rawVersion;
-    }
+    private static final Pattern VERSION_PATTERN = Pattern.compile( "(\\d+\\.)+(\\d+)" );
 
-    public static GpgVersion parse( String rawVersion )
+    private final int[] versionSegments;
+
+    private GpgVersion( int... versionSegments )
     {
-        return new GpgVersion( rawVersion );
+        this.versionSegments = versionSegments;
     }
 
-    @Override
-    public int compareTo( GpgVersion other )
+    public static GpgVersion parse( String rawVersion )
     {
-        Pattern p = Pattern.compile( "(\\d+\\.)+(\\d+)" );
-
-        String[] thisSegments;
-        Matcher m = p.matcher( rawVersion );
-        if ( m.find() )
+        final Matcher versionMatcher = VERSION_PATTERN.matcher( rawVersion );
```

Comment:
The regex might be overkill. Looks like you can just catch and convert the number format exception below.

Reply:
The original author implemented it in a way, that the full output of `gpg --version` (e.g. `gpg (GnuPG) 2.2.1` or `gpg (GnuPG) 2.0.26 (Gpg4win 2.2.3)`) is passed to this function.
They used the regex to extract the version part out of that output.

It is debatable whether that is the best way to extract the version number (personally I would have extracted the version number part already in [GpgVersionParser.GpgVersionConsumer](https://github.com/apache/maven-gpg-plugin/blob/master/src/main/java/org/apache/maven/plugins/gpg/GpgVersionParser.java#L113) before passing it in here), but I would like to avoid introducing new bugs by breaking with the existing external contract of this class.

In the end it's your call again and I could widen the scope of the refactoring to the whole GPG version parser.